pkg/volume/csi/csi_attacher.go

```diff
@@ -269,6 +269,25 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		return err
 	}
 
+	dataDir := filepath.Dir(deviceMountPath)
+	if err := os.MkdirAll(dataDir, 0750); err != nil {
+		glog.Error(log("attacher.MountDevice failed to create dir %#v:  %v", dataDir, err))
+		return err
+	}
+
+	data := map[string]string{
+		volDataKey.volHandle:  csiSource.VolumeHandle,
+		volDataKey.driverName: csiSource.Driver,
```

Comment:
The mount side saves a lot more:

```GO
		volDataKey.specVolID:    spec.Name(),
		volDataKey.volHandle:    pvSource.VolumeHandle,
		volDataKey.driverName:   pvSource.Driver,
		volDataKey.nodeName:     node,
		volDataKey.attachmentID: attachID,
```

Are we sure we don't need anything else from the spec?

Reply:
No, [UnstageVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#nodeunstagevolume) needs just volume id (+ driver).

Reply:
@saad-ali the mount side saves more info because it's needed to reconstruct the VolumeAttachment ID in subsequent steps.